usvm-go/src/main/kotlin/org/usvm/machine/GoMachine.kt

```diff
@@ -0,0 +1,129 @@
+package org.usvm.machine
+
+import mu.KLogging
+import org.usvm.StateCollectionStrategy
+import org.usvm.UMachine
+import org.usvm.UMachineOptions
+import org.usvm.bridge.GoBridge
+import org.usvm.machine.interpreter.GoInterpreter
+import org.usvm.machine.state.GoMethodResult
+import org.usvm.machine.state.GoState
+import org.usvm.ps.createPathSelector
+import org.usvm.statistics.*
+import org.usvm.statistics.collectors.CoveredNewStatesCollector
+import org.usvm.statistics.collectors.TargetsReachedStatesCollector
+import org.usvm.statistics.distances.CallGraphStatisticsImpl
+import org.usvm.statistics.distances.CfgStatisticsImpl
+import org.usvm.statistics.distances.PlainCallGraphStatistics
+import org.usvm.stopstrategies.createStopStrategy
+
+val logger = object : KLogging() {}.logger
+
+class GoMachine(
+    private val options: UMachineOptions,
+) : UMachine<GoState>() {
+    private val bridge = GoBridge()
+    private val typeSystem = GoTypeSystem(bridge, options.typeOperationsTimeout)
+    private val applicationGraph = GoApplicationGraph(bridge)
+    private val components = GoComponents(typeSystem, options)
+    private val ctx = GoContext(components)
+    private val interpreter = GoInterpreter(bridge, ctx)
+    private val cfgStatistics = CfgStatisticsImpl(applicationGraph)
+
+    override fun close() {
+        ctx.close()
+    }
+
+    fun analyze(file: String, entrypoint: String, debug: Boolean): List<GoState> {
+        bridge.initialize(file, debug)
+
+        val entryPoint = bridge.getMethod(entrypoint)
+        val startCode = bridge.start()
+        logger.debug("Bridge start: code {}", startCode)
+        if (startCode != 0) {
+            return emptyList()
+        }
+
+        val results = analyze(listOf(entryPoint))
+        val shutdownCode = bridge.shutdown()
+        logger.debug("Bridge shutdown: code {}", shutdownCode)
```

Comment:
Looks odd that you print "start" without ending if there was a non-zero start code